rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/NSCreationServlet.java

```diff
@@ -159,55 +160,71 @@ private void createNs(HttpServletRequest request, HttpServletResponse response)
+                                                                                             toArray(infraFileParams),
+                                                                                             policy,
+                                                                                             toArray(policyParams),
+                                                                                             toArray(policyFileParams),
+                                                                                             nodesRecoverable);
             }
 
-            if (deployNodeSource) {
-                jsonResult = ((RMServiceImpl) RMServiceImpl.get()).deployNodeSource(sessionId, nsName);
+            JSONObject jsonObject = new JSONObject(jsonResponsePayload);
+            if (hasResult(jsonObject) && isResultSuccessful(jsonObject)) {
+                jsonResponsePayload = createJsonPair(jsonObject);
+            } else {
+                deployNodeSource = false;
             }
 
-            JSONObject json = new JSONObject(jsonResult);
-            if (json != null) {
-                if (json.has("result")) {
-                    if (json.getBoolean("result")) {
-                        jsonResult = createNonEscapedSimpleJsonPair("result", "true");
-                    } else {
-                        String errorMessage = json.get("errorMessage").toString();
-                        write(response,
-                              createJavascriptPayload(callbackName,
-                                                      createEscapedSimpleJsonPair("errorMessage", errorMessage)));
-                        return;
+            if (deployNodeSource) {
+                jsonResponsePayload = ((RMServiceImpl) RMServiceImpl.get()).deployNodeSource(sessionId, nsName);
 
-                    }
+                jsonObject = new JSONObject(jsonResponsePayload);
+                if (hasResult(jsonObject)) {
+                    jsonResponsePayload = createJsonPair(jsonObject);
                 }
             }
-            write(response, createJavascriptPayload(callbackName, jsonResult));
+
+            write(response, createJavascriptPayload(callbackName, jsonResponsePayload));
         } catch (Throwable t) {
             write(response,
                   createJavascriptPayload(callbackName, createEscapedSimpleJsonPair("errorMessage", t.getMessage())));
         }
     }
 
+    private String createJsonPair(JSONObject json) throws JSONException {
+        String jsonResult;
+        if (isResultSuccessful(json)) {
+            jsonResult = createNonEscapedSimpleJsonPair("result", "true");
+        } else {
+            String errorMessage = json.get("errorMessage").toString();
+            jsonResult = createEscapedSimpleJsonPair("errorMessage", errorMessage);
+        }
+        return jsonResult;
+    }
+
+    private boolean hasResult(JSONObject jsonObject) {
```

Comment:
Maybe you don't need this method because you don't use it to throw any error but you just pair it with `isResultSuccessful()`, so I think you can insert its tests in the later (`isResultSuccessful()`)
It's up to you

Reply:
It is actually used at two different places, once in combination with `isResultSuccessful` and once without the `isResultSuccessful` check. So I will keep `hasResult` as a method.

Reply:
@jrochas Oh, sorry, I didn't notice that, then you're right ;) thx